extension.js

```diff
@@ -215,22 +216,44 @@ const BingWallpaperIndicator = new Lang.Class({
         if (this._updatePending)
             return;
         this._updatePending = true;
-
         this._restartTimeout();
-
         let market = this._settings.get_string('market');
-        log("market: " + market);
-
         // create an http message
-        let request = Soup.Message.new('GET', BingImageURL+market); // + market
-        log("fetching: " + BingImageURL+market);
-
+        let request = Soup.Message.new('GET', BingImageURLPre+market); // + market
+        //notify("Check Bing.com. For "+BingImageURLPre, BingImageURLPre+market,true);
         // queue the http request
         httpSession.queue_message(request, Lang.bind(this, function(httpSession, message) {
             if (message.status_code == 200) {
                 let data = message.response_body.data;
                 log("Recieved "+data.length+" bytes");
-                this._parseData(data);
+                //notify("Get Json Data. "+message.status_code, data,true);
+                let checkData=JSON.parse(data);
+                let checkStatus=checkData['market']['mkt'];
+                BingImageURL=BingURL+checkData['images'][0]['url'];
+                if(checkStatus == market){
+                   this._parseData(checkData);
+                }else {
+                  market="en-ww";//if request value of mkl is not the user wants. Used en-ww instead.
+                  // create an http message by market with en-ww.
+                  httpSession.queue_message(Soup.Message.new('GET', BingBaseImageURL), Lang.bind(this, function(httpSession, message) {
+                      if (message.status_code == 200) {
```

Comment:
Just to confirm, the status received is always 200? Not 203 or 30x when the market doesn't match?

Reply:
Maybe 203 or 30x need your help to decide what to do. Do a notify or try to use another request URL. These may be an important decision.

Reply:
If you run this command:
`curl -D - https://www.bing.com/HPImageArchive.aspx\?format\=js\&idx\=0\&n\=1\&mbl\=1\&mkt\=en-US`
Does it return this as the first line?
`HTTP/1.1 200 OK`

Reply:
yes, got 200.

![record info](https://raw.githubusercontent.com/gwdgithubnom/issues-resources/master/neffo/bing-wallpaper-gnome-extension/2017/10/02/2017-10-02-22-30.gif)